lib/Classifiers/Audio/MusicnnClassifier.php

```diff
@@ -7,116 +7,68 @@
-
-		$command = [
-			$this->config->getAppValue('recognize', 'node_binary'),
-			dirname(__DIR__, 3) . '/src/classifier_musicnn.js',
-			'-'
-		];
-
-		$this->logger->debug('Running '.var_export($command, true));
-		$proc = new Process($command, __DIR__);
-		if ($this->config->getAppValue('recognize', 'tensorflow.gpu', 'false') === 'true') {
-			$proc->setEnv(['RECOGNIZE_GPU' => 'true']);
-		}
+		}, $audios);
 		if ($this->config->getAppValue('recognize', 'tensorflow.purejs', 'false') === 'true') {
-			$proc->setEnv(['RECOGNIZE_PUREJS' => 'true']);
-			$proc->setTimeout(count($paths) * self::FILE_PUREJS_TIMEOUT + self::MODEL_DOWNLOAD_TIMEOUT);
+			$timeout = count($paths) * self::FILE_PUREJS_TIMEOUT + self::MODEL_DOWNLOAD_TIMEOUT;
 		} else {
-			$proc->setTimeout(count($paths) * self::FILE_TIMEOUT + self::MODEL_DOWNLOAD_TIMEOUT);
+			$timeout = count($paths) * self::FILE_TIMEOUT + self::MODEL_DOWNLOAD_TIMEOUT;
 		}
-		$proc->setInput(implode("\n", $paths));
-		try {
-			$proc->start();
-
-			// Set cores
-			$cores = $this->config->getAppValue('recognize', 'tensorflow.cores', '0');
-			if ($cores !== '0') {
-				@exec('taskset -cp ' . implode(',', range(0, (int)$cores, 1)) . ' ' . $proc->getPid());
-			}
 
-			$i = 0;
-			$errOut = '';
-			foreach ($proc as $type => $data) {
-				if ($type !== $proc::OUT) {
-					$errOut .= $data;
-					$this->logger->debug('Classifier process output: '.$data);
-					continue;
-				}
-				$lines = explode("\n", $data);
-				foreach ($lines as $result) {
-					if (trim($result) === '') {
-						continue;
-					}
-					$this->logger->debug('Result for ' . $files[$i]->getName() . ' = ' . $result);
-					try {
-						// decode json
-						$tags = json_decode($result, true, 512, JSON_OBJECT_AS_ARRAY | JSON_THROW_ON_ERROR | JSON_INVALID_UTF8_IGNORE);
+		// Call out to node.js process
+		$classifierProcess = $this->classifyFiles(self::MODEL_NAME, $paths, $timeout);
 
+		foreach ($classifierProcess as $i => $results) {
+			// assign tags
+			$this->tagManager->assignTags($audios[$i]->getFileId(), $results);
 
-						// assign tags
-						$this->tagManager->assignTags($files[$i]->getId(), $tags);
-					} catch (InvalidPathException $e) {
-						$this->logger->warning('File with invalid path encountered');
-					} catch (NotFoundException $e) {
-						$this->logger->warning('File to tag was not found');
-					} catch (\JsonException $e) {
-						$this->logger->warning('JSON exception');
-						$this->logger->warning($e->getMessage());
-						$this->logger->warning($result);
-					}
-					$i++;
-				}
-			}
-			if ($i !== count($files)) {
-				$this->logger->warning('Classifier process output: '.$errOut);
-				throw new \RuntimeException('Classifier process error');
+			// Update processed status
+			$audios[$i]->setProcessedMusicnn(true);
+			try {
+				$this->audioMapper->update($audios[$i]);
+			} catch (Exception $e) {
+				$this->logger->warning($e->getMessage());
```

Comment:
Maybe a general thing. You can add the exception through the logger interface:

```suggestion
				$this->logger->warning($e->getMessage(), ['exception' => $e]);
```